content/en/docs/tasks/administer-cluster/migrating-from-dockershim/find-out-runtime-you-use.md

```diff
@@ -49,4 +52,33 @@ node-3       Ready    v1.19.6   containerd://1.4.1
 ```
 
 Find out more information about container runtimes
-on [Container Runtimes](/docs/setup/production-environment/container-runtimes/) page.
+on [Container Runtimes](/docs/setup/production-environment/container-runtimes/)
+page.
+
+## Find out what container runtime endpoint you use {#which-endpoint}
+
+The container runtime talks to the kubelet uver a Unix socket using the gRPC
+framework. The kubelet acts as a client, and the runtime acts as the server.
+In some cases, you might find it useful to know which socket your nodes use. For
+example, with the removal of dockershim in Kubernetes 1.24 and later, Docker
+Engine cannot communicate with the kubelet if it listens on the
+`unix:///var/run/dockershim.sock` endpoint. 
+
+{{<note>}}
+If you currently use Docker Engine in your nodes with `cri-dockerd`, you aren't
+affected by the dockershim removal.
+{{</note>}}
+
+You can check which socket you use by checking the kubelet configuration on your
+nodes. 
+
+For example, in a cluster created using the `kubeadm` deployment tool, do the
+following:
+
+1.  Open `/var/lib/kubelet/kubeadm-flags.env`.
+1.  Look for the `--container-runtime-endpoint` flag. If you don't see the flag,
+    you're probably using `dockershim`. In this case, [Migrate to another
+    runtime](/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd) or [Migrate to cri-dockerd](/tasks/migrate-dockershim-dockerd).
```

Comment:
If the kubelet is running, you could look at the command line that it started with. You could also look at things like systemd units.

I'd probably do that even in the presence of a `/var/lib/kubelet/kubeadm-flags.env` because I like to get authoritative info.